src/naksu/network/network_linux.go

```diff
@@ -57,12 +57,8 @@ func getExtInterfaceSpeed(extInterface string) uint64 {
 
 	/* #nosec */
 	carrierFileContent, err := ioutil.ReadFile(carrierPath)
-	if err != nil {
-		log.Debug(fmt.Sprintf("Could not read network interface carrier from '%s': %v", carrierPath, err))
-		return 0
-	}
-	if strings.TrimSpace(string(carrierFileContent)) == "0" {
-		return 0 // Link not up
+	if err != nil || strings.TrimSpace(string(carrierFileContent)) == "0" {
+		return 0 // Network interface is powered down or disconnected from the network.
```

Comment:
Mainitsisin tässä ehkä vielä ton https://bugzilla.redhat.com/show_bug.cgi?id=1253797 -bugin

Reply:
Sehän siis ei ole bugi, vaan bugirapsa jonka kommenteista selviää, että tuo on odotettu toiminnallisuus. Saman saa selville muistakin lähteistä, jotka käsittelevät sysfs:n carrierin toimintaa. Vai tarkoitatko että ylipäätään olisi hyvä linkittää jokin lähde tämän toiminnasta?

Reply:
Olennaista ois varmaan mainita, että `err != nil` voi seurata siitä, että linkki on alhaalla. Jos ois oikein noheva, niin vois tsekkaa, et onko kyseessä toi "invalid argument"-virhe omassa if-haarassaan, ja siihen kommentoida et toi invalid argument voi tulla, jos linkki on alhaalla, koska virransäästö tms.

Reply:
Ja lokittaa odottamattomat virhetilanteet (= ei toi invalid argument).

Reply:
Mjoo. Pidän epätodennäköisinä näitä odottamattomia virhetilanteita, mutta voidaanhan ne lokittaa.

Err != nil seuraa siis nimenomaan siitä, että verkkolaiteesta on virrat pois, minkä vuoksi siltä ei voida kysellä tietoja. Tämä lukee commit-viestissä. Mun järkeily tuossa kommentissa oli, että siinä on sama rakenne kuin ehtolausekkeessa (`err != nil` => powered down tai sisältö on 0 => kaapeli irti tms.). Voin kirjoittaa verboosimmin.

Reply:
Olisko nyt riittävän verboosia? Sain jopa testattua kun oli langaton verkkokortti käyttämättä. Aikamoista.

Reply:
No nyt on :chefs-kiss:-tason koodia, :maisc:!